app/src/org/commcare/dalvik/activities/AppManagerActivity.java

```diff
@@ -0,0 +1,167 @@
+
+import org.commcare.android.adapters.AppManagerAdapter;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.R;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.services.CommCareSessionService;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.ListView;
+import android.widget.Toast;
+
+/**
+ * The activity that starts up when a user launches into the app manager.
+ * Displays a list of all installed apps, each of which can be clicked to launch
+ * the SingleAppManagerActivity for that app. Also includes a button for
+ * installing new apps.
+ *
+ * @author amstone326
+ */
+
+public class AppManagerActivity extends Activity implements OnItemClickListener {
+
+    public static final String KEY_LAUNCH_FROM_MANAGER = "from_manager";
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.app_manager);
+        ((ListView)this.findViewById(R.id.apps_list_view)).setOnItemClickListener(this);
+    }
+
+    public void onResume() {
+        super.onResume();
+        refreshView();
+    }
+
+    /**
+     * Refresh the list of installed apps
+     */
+    private void refreshView() {
+        ListView lv = (ListView)findViewById(R.id.apps_list_view);
+        lv.setAdapter(new AppManagerAdapter(this, android.R.layout.simple_list_item_1,
+                CommCareApplication._().appRecordArray()));
+    }
+
+    /**
+     * onClick method for the Install An App button
+     *
+     * @param v unused argument necessary for the method's use as an onClick handler.
+     */
+    public void installAppClicked(View v) {
+        try {
+            CommCareSessionService s = CommCareApplication._().getSession();
+            if (s.isActive()) {
+                triggerLogoutWarning();
+            } else {
+                installApp();
+            }
+        } catch (SessionUnavailableException e) {
+            installApp();
+        }
+    }
+
+    /**
+     * Logs the user out and takes them to the app installation activitiy.
+     */
+    private void installApp() {
+        try {
+            CommCareApplication._().getSession().closeSession(false);
+        } catch (SessionUnavailableException e) {
+            // If the session isn't available, we don't need to logout
+        }
+        Intent i = new Intent(getApplicationContext(), CommCareSetupActivity.class);
+        i.putExtra(KEY_LAUNCH_FROM_MANAGER, true);
+        this.startActivityForResult(i, CommCareHomeActivity.INIT_APP);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
```

Comment:
Could use a comment detailing that this both launches the media validation activity and processes its result.